### PR TITLE
make sure fromtimestamp always return FakeDateTime

### PR DESCRIPTION
**Pull request by Koswu**
This fix https://github.com/spulec/freezegun/issues/344

This is my first PR, if something goes wrong, please let me know, PTAL

### CODE REVIEW

**Comment by boxed** [comment]
The proposed code change seems to be no change of behavior. Could you point out what it changes?

**Comment by Koswu** [comment]
In the `fromtimestamp` method, when the `tz` parameter is None, It will return the real datetime object. This object's `.timestamp()` behavior will [depend on the system timezone](https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp), even it tzinfo is None.  For example: 
```
# my system timezone is UTC+8
>>> dt = datetime.datetime(1970, 1, 1, 8, 0, 1)
>>> dt.tzinfo
None
>>> dt.timestamp()
1.0
```

However, the fake one depend on the freeze time zone: 
```
>>> freezer = freeze_time(datetime.datetime(1970, 1, 1, 0, 0, 1), tz_offset=-1)
>>> freezer.start()
>>> datetime.datetime(1970, 1, 1, 8, 0, 1).timestamp()
32401.0
``` 

So if we make these steps:
1. make a fake datetime object
2. convert it to timestamp
3. convert it to datetime
4. convert it to timestamp

we will have a type chain like this: 
>  FakeDatetime -> int -> RealDatetime -> int

Since we know there are different behavior between `FakeDatetime.timestamp()` and `RealDatetime.timestamp()`, it will cause problem that two timestamp in the type chain are not equal. 

That's why I added the test.

File: freezegun/api.py [hunk 1]
```diff
@@ -371,10 +371,11 @@ def astimezone(self, tz=None):
...
     def fromtimestamp(cls, t, tz=None):
         if tz is None:
-            return real_datetime.fromtimestamp(
-                    t, tz=dateutil.tz.tzoffset("freezegun", cls._tz_offset())
-                ).replace(tzinfo=None)
-        return datetime_to_fakedatetime(real_datetime.fromtimestamp(t, tz))
+            tz = dateutil.tz.tzoffset("freezegun", cls._tz_offset())
+            result = real_datetime.fromtimestamp(t, tz=tz).replace(tzinfo=None)
+        else:
+            result = datetime_to_fakedatetime(real_datetime.fromtimestamp(t, tz))
```
Comment (bblommers) [inline freezegun/api.py:377]:
```suggestion
            result = real_datetime.fromtimestamp(t, tz)
```

The `datetime_to_fakedatetime` already happens at the end - no need to duplicate this call.

Happy to merge once that's fixed.

**Review by bblommers — CHANGES_REQUESTED** [review]

File: freezegun/api.py (hunk 1 above)
Comment (Koswu) [reply freezegun/api.py]: fixed. thanks for pointing this out.

**Review by bblommers — APPROVED** [review]
Great - thanks for the fix @Koswu!

**Pull request merged**